controllers/space/space_controller.go

```diff
@@ -0,0 +1,343 @@
+	// create if not found
+	nsTmplSet := &toolchainv1alpha1.NSTemplateSet{}
+	if err := memberCluster.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: memberCluster.OperatorNamespace,
+		Name:      space.Name,
+	}, nsTmplSet); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("creating NSTemplateSet on target member cluster")
+			if err := r.setStatusProvisioning(space); err != nil {
+				return false, err
+			}
+			if nsTmplSet, err = r.newNSTemplateSet(memberCluster.OperatorNamespace, space); err != nil {
+				return false, err
+			}
+
+			if err := memberCluster.Client.Create(context.TODO(), nsTmplSet); err != nil {
+				logger.Error(err, "failed to create NSTemplateSet on target member cluster")
+				return false, r.setStatusNSTemplateSetCreationFailed(space, err.Error())
+			}
+			logger.Info("NSTemplateSet created on target member cluster")
+			return true, nil
+		}
+		return false, r.wrapErrorWithStatusUpdate(logger, space, r.setStatusNSTemplateSetCreationFailed, err, "failed to get NSTemplateSet '%s'", name)
+	}
+	logger.Info("NSTemplateSet already exists")
+
+	readyCond, found := condition.FindConditionByType(nsTmplSet.Status.Conditions, toolchainv1alpha1.ConditionReady)
+	// if UserAccount ready condition is already set to false, then don't update if no message is provided
+	// also, don't update when NSTemplateSet ready condition is in an invalid state
+	if !found || readyCond.Status == corev1.ConditionUnknown || readyCond.Status == corev1.ConditionFalse {
+		logger.Info("NSTemplateSet is not ready", "ready-condition", readyCond)
+		if err := r.setStatusFromNSTemplateSet(space, readyCond.Reason, readyCond.Message); err != nil {
+			return false, err
+		}
+		return true, nil
+	}
+	return false, r.setStatusReady(space)
+}
+
+func (r *Reconciler) newNSTemplateSet(memberOperatorNS string, space *toolchainv1alpha1.Space) (*toolchainv1alpha1.NSTemplateSet, error) {
+	// look-up the NSTemplateTier
+	tmplTier := &toolchainv1alpha1.NSTemplateTier{}
+	if err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: space.Namespace,
+		Name:      space.Spec.TierName,
+	}, tmplTier); err != nil {
+		return nil, err
+	}
+	// create the NSTemplateSet from the NSTemplateTier
+	nsTmplSet := &toolchainv1alpha1.NSTemplateSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: memberOperatorNS,
+			Name:      space.Name,
+			Finalizers: []string{
+				toolchainv1alpha1.FinalizerName,
+			},
+		},
+		Spec: toolchainv1alpha1.NSTemplateSetSpec{
+			TierName: space.Spec.TierName,
+			ClusterResources: &toolchainv1alpha1.NSTemplateSetClusterResources{
+				TemplateRef: tmplTier.Spec.ClusterResources.TemplateRef,
+			},
+			Namespaces: make([]toolchainv1alpha1.NSTemplateSetNamespace, len(tmplTier.Spec.Namespaces)),
+		},
+	}
+	for i, tmpl := range tmplTier.Spec.Namespaces {
+		nsTmplSet.Spec.Namespaces[i] = toolchainv1alpha1.NSTemplateSetNamespace{
+			TemplateRef: tmpl.TemplateRef,
+		}
+	}
```

Comment:
Could you please use the already existing function https://github.com/codeready-toolchain/host-operator/blob/master/controllers/usersignup/masteruserrecord.go#L109-L127
We should be sure that it will create the very same content of the NSTempalteSet  CRs

Reply:
sure! done in 36a81e1